packages/react-notion-x/src/third-party/collection-view-calendar.tsx

```diff
@@ -0,0 +1,436 @@
+              <div className='notion-calendar-header-days-day'>Tue</div>
+              <div className='notion-calendar-header-days-day'>Wed</div>
+              <div className='notion-calendar-header-days-day'>Thu</div>
+              <div className='notion-calendar-header-days-day'>Fri</div>
+              <div className='notion-calendar-header-days-day'>Sat</div>
+              <div className='notion-calendar-header-days-day'>Sun</div>
+            </>
+          ) : (
+            <>
+              <div className='notion-calendar-header-days-day'>Sun</div>
+              <div className='notion-calendar-header-days-day'>Mon</div>
+              <div className='notion-calendar-header-days-day'>Tue</div>
+              <div className='notion-calendar-header-days-day'>Wed</div>
+              <div className='notion-calendar-header-days-day'>Thu</div>
+              <div className='notion-calendar-header-days-day'>Fri</div>
+              <div className='notion-calendar-header-days-day'>Sat</div>
+            </>
+          )}
+        </div>
+      </div>
+
+      <div
+        style={{
+          height: '66px'
+        }}
+      ></div>
+
+      <div className='notion-calendar-body'>
+        {weeksArr.map((i, indexI) => (
+          <div
+            className={cs(
+              'notion-calendar-body-inner',
+              weeksArr.length - 1 == indexI &&
+                'notion-calendar-body-inner-last-week'
+            )}
+            style={{
+              height: `${
+                collectionView.format?.calendar_properties &&
+                Object.keys(collectionView.format?.calendar_properties).length *
+                  20 +
+                  64
+              }px`
+            }}
+            key={indexI}
+          >
+            {i.dates.map((day, indexY) => (
+              <>
+                <div
+                  className={cs(
+                    'notion-selectable',
+                    'notion-calendar-body-inner-week',
+                    startWeekOnMonday
+                      ? indexY == 5 || indexY == 6
+                        ? 'notion-calendar-body-inner-week-dif'
+                        : ''
+                      : indexY == 0 || indexY == 6
+                      ? 'notion-calendar-body-inner-week-dif'
+                      : ''
+                  )}
+                  key={indexY}
+                >
+                  <div
+                    className={cs(
+                      'notion-calendar-body-inner-day',
+                      day == new Date(Date.now()).getDate() &&
+                        currentYear.getMonth() ==
+                          new Date(Date.now()).getMonth() &&
+                        currentYear.getFullYear() ==
+                          new Date(Date.now()).getFullYear()
+                        ? 'notion-calendar-body-inner-day-today'
+                        : (day == 1 && currentMonth++ == 0) ||
```

Comment:
Modifying variables directly inside of JSX is a big red flag. One of the core invariants of React is that React may invoke the render function of a component as many times as it wants and it should expect the same output as long as its inputs haven't changed (state + context).

I'm not totally following the logic here, but in any case, `currentMonth++` is modifying state that the render function relies on and needs to go.